spec/classes/service_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+
+describe 'puppetwebhook::service' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+      let(:pre_condition) { 'include puppetwebhook' }
+
+      it { is_expected.to compile }
+      it { is_expected.to contain_class('puppetwebhook') }
+      it { is_expected.to contain_class('puppetwebhook::service') }
+      it { is_expected.to contain_systemd__unit_file('puppet_webhook.service').with_ensure('present') }
```

Comment:
Is it worth testing the templated content? Perhaps since it’s epp, it’s already less likely to be wrong. (Eg the template is validating its params).

Reply:
I was planning on testing the content in the acceptance tests, unless there is an `rspec-puppet` framework for testing `epp` files. Otherwise, ensuring the file itself generates correctly is more useful here.